### ISSUE TITLE
Per-page cop documentation

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#13886]
Ref: https://github.com/rubocop/rubocop/issues/13860#issuecomment-2671441769

Currently, all cops of a department are put on a single page. This is annoying:
* If you navigate to that page from search engine results, you will be put at the top and have to then search for that cop on the page again
* SEO is bad (maybe, not sure). I'm somewhat certain this still has at least some negative impact.

It should be pretty easy to change this if desired but backwards compatibility should be considered. Doc links from `Cop.documentation_url` are not versioned, so tools like `ruby-lsp` always link to the latest version. If the structure changes, all older rubocop versions would 404, so it should at least be kept around and refer to the new location.

I will explore how this could look in the coming days.

**Comment by bbatsov** [comment]
> If the structure changes, all older rubocop versions would 404, so it should at least be kept around and refer to the new location.

Yeah, I'm guessing that we can just have links to the new pages in the department pages in that would be more or less OK. 

We haven't been paying that much attention to the SEO so far - probably the fact that we're deleting older versions of the versioned docs from time to time doesn't help much for it either. 

I tried to use https://docs.antora.org/antora/latest/playbook/urls-latest-version-segment/ in the past, but for some reason I haven't been able to get it to work. (basically it would ensure that there's a stable name for the latest stable version docs)

**Comment by Earlopain** [comment]
I'm not sure how much of an issue keeping the old versions around actually is. For Rails, I consistently get results for ancient versions like 4.2 when searching, but I haven't observed the same for RuboCop. I don't think imagine they are linked to from many different places so I think the impact should be pretty low. _But_, I'm not a SEO guru.

Anyways, Rails improved this via https://github.com/rails/rails/pull/53811 and something similar should apply to RuboCop as well. `<link rel="canonical" href="">` tells search engines what the "real" version is, and that could be added to the docs. So, `https://docs.rubocop.org/rubocop/1.72/index.html` would tell that `https://docs.rubocop.org/rubocop/index.html` should be important to search engines.

**Comment by Earlopain** [comment]
I experimented a bit with this and am no longer convinced that this is the right thing to do.

Currently, it's very nice for discoverability for so many cops to be on the same page. At least if you're already on the page and aren't looking for a specific one, you can just scroll through the list and read about them. If every one would get its own page that's no longer possible, which (at least to me) is enough to take a step back from this idea.

**Comment by koic** [comment]
Manually following hyperlinks each time is inconvenient, and providing users with an overview of cops is a useful use case. It would be practical to prepare a page with permalinks for individual cops and link to each cop from the current department-specific cop list pages. This approach maintains the existing page structure while simply adding new cop-specific page extensions, ensuring compatibility.

**Comment by koic** [comment]
Here is a consideration for preparing individual pages. The permalink would be based on the cop name, but if the cop name changes or is moved to a different department, the permalink would also change. Ideally, instead of resulting in a 404 Not Found, the web should redirect (e.g., based on `config/obsoletion.yml`) .

**Comment by bbatsov** [comment]
> Manually following hyperlinks each time is inconvenient, and providing users with an overview of cops is a useful use case. It would be practical to prepare a page with permalinks for individual cops and link to each cop from the current department-specific cop list pages. This approach maintains the existing page structure while simply adding new cop-specific page extensions, ensuring compatibility.

I was about to suggest something similar.

**Comment by Earlopain** [comment]
Sorry, I'm not 100% following. I understand this as keeping the current documentation mostly as is, and adding single-cop pages in addition, is that right?

And make the list of cops link there:
![grafik](https://github.com/user-attachments/assets/d3027502-2621-4254-b068-918196cf2810)

**Comment by koic** [comment]
Yes. At the HTML level, it should be sufficient to add hyperlinks to individual cop pages to each cop entry in the existing list page:

```html
[<a href=https://docs.rubocop.org/rubocop/1.72/Lint/AmbiguousAssignment.html>permalink</a>]
```

**Comment by Earlopain** [comment]
This makes sense to me, thanks.

One question about antora. I cloned the docs repo and successfully build it but am uncertain on how to have it pick up my local changes to preview. I changed the source from the git repo to `url: ../rubocop` and rebuild but it doesn't seem to reflect any changes I make. Any idea on what I'm doing wrong?

**Comment by bbatsov** [comment]
Did you try `antora --fetch`? This should force it to update its repo cache.
